examples/ticketing-connector/main.go

```diff
@@ -0,0 +1,521 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"strings"
+
+	"github.com/google/uuid"
+
+	"github.com/synqly/go-sdk/client/engine"
+	engineClient "github.com/synqly/go-sdk/client/engine/client"
+	mgmt "github.com/synqly/go-sdk/client/management"
+	mgmtClient "github.com/synqly/go-sdk/client/management/client"
+)
+
+var consoleLogger = log.New(os.Stdout, "INFO: ", log.Ldate|log.Ltime)
+
+type App struct {
+	Tenant *Tenant
+	Config *Config
+}
+
+// NewApp instantiates a new App
+func NewApp(config *Config) *App {
+	return &App{
+		Config: config,
+	}
+}
+
+// A Tenant object represents one of your customers, as well as the state you
+// would maintain for them in your application.
+type Tenant struct {
+	// ID: A unique identifier for one of your customers.
+	ID string
+	// SynqlyAccountId: The ID of the Synqly Account in which Integrations for
+	// this tenant will be created. This example creates a new Account for every
+	// Tenant, but it would also be possible to use the same Account for all Tenants.
+	SynqlyAccountId string
+	// SynqlyClient: A cached management client, used to manage Integrations.
+	SynqlyClient *mgmtClient.Client
+	// TicketClient: A cached engine client, used to log events to a third-party
+	// logging Provider by way of an Integration.
+	TicketClient *engineClient.Client
+}
+
+// NewTenant initializes a Synqly Account for a given Tenant. This example
+// creates a new Account for every Tenant to keep their credentials isolated.
+//
+// Returns an error if a tenant with the same ID already exists or if a Synqly
+// Account cannot be created for the Tenant.
+func (a *App) NewTenant(ctx context.Context, id string) error {
+	// Do not allow duplicate tenant names
+
+	if a.Tenant != nil && a.Tenant.ID == id {
+		return fmt.Errorf("duplicate tenant name %v", id)
+	}
+
+	// Create a Synqly Client that can be used to interact with the tenant
+	// We will use this client to create an Account and set up an Integration with an event logging provider
+	client := mgmtClient.NewClient(
+		mgmtClient.WithToken(a.Config.SynqlyOrgToken),
+	)
+
+	// Create a Synqly Account for this tenant
+	account, err := client.Accounts.Create(ctx, &mgmt.CreateAccountRequest{
+		Fullname: &id,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to create account: %w", err)
+	}
+
+	// Store the Tenant and associated Synqly objects in an in-memory cache.
+	a.Tenant = &Tenant{
+		ID:              id,
+		SynqlyAccountId: account.Result.Account.Id,
+		SynqlyClient:    client,
+		TicketClient:    nil,
+	}
```

Comment:
can you update this to remove `SynqlyAccountId` from the Tenant struct? instead, create the account like

```
	account, err := client.Accounts.Create(ctx, &mgmt.CreateAccountRequest{
		Name: &id,
	})
```

you will need to add validation that the `id` here contains only `[A-Za-z]` and `-` or `_` though. In Synqly, names can be used directly instead of the id. Then when you delete the account in `cleanup`, you can do 

```
	if err := app.Tenant.SynqlyClient.Accounts.Delete(ctx, app.Tenant.ID); err != nil {
		consoleLogger.Printf("Error deleting account %s: %s\n", app.Tenant.SynqlyAccountId, err)
	}
```